File: leetcode-java/src/main/java/chapter03/BinaryTreeInvert.java
package chapter03;

import utils.TreeNode;

/**
 * 翻转二叉树
 */
public class BinaryTreeInvert {

    public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return null;
        }
        TreeNode left = invertTree(root.left);
        TreeNode right = invertTree(root.right);
        root.left = right;
        root.right = left;
        return root;
    }

}

File: leetcode-go/chapter03/binary_tree_invert.go
package chapter03

import nodes "leetcode-go/node"

// 翻转二叉树
type BinaryTreeInvert struct {}

// 递归实现
func (tree *BinaryTreeInvert) InvertTree(root *nodes.TreeNode) *nodes.TreeNode {
	if root == nil {
		return nil
	}
	left := tree.InvertTree(root.Left)
	right := tree.InvertTree(root.Right)
	root.Left = right
	root.Right = left
	return root
}